src/App.tsx

```diff
@@ -1,8 +1,76 @@
+/* eslint-disable no-console */
 import React from 'react';
 import './App.css';
+import { GoodsList } from './GoodsList';
+import { getGoods } from './api';
+import { GoodProps } from './interfaces';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface State {
+  isLoading: boolean;
+  goods: GoodProps[];
```

Comment:
почему Props? как вы придумываете такие названия)
```suggestion
  goods: GoodInterface[];
```